packages/starlight/components/SiteTitle.astro

```diff
@@ -23,7 +23,8 @@ if (config.logo) {
   if (err) throw new Error(err);
 }
 
-const href = pathWithBase(Astro.props.locale || '/');
+const locale =  Astro.props.locale === 'root' ? '' : Astro.props.locale;
+const href = pathWithBase(locale || '/');
```

Comment:
I think we can safely revert this here:
```suggestion
const href = pathWithBase(Astro.props.locale || '/');
```
`locale` is really a property of the route, so if this fails here, it should be fixed elsewhere. And `SiteTitle` is only used in one place, so again, if we get a bad value here, chances are there’s a bad value somewhere else too.